stumpy/core.py

```diff
@@ -1238,6 +1245,9 @@ def mass_absolute(Q, T, T_subseq_isfinite=None, p=2.0):
         raise ValueError(f"`Q` is {Q.ndim}-dimensional and must be 1-dimensional. ")
 
     check_window_size(m, max_size=Q.shape[-1])
+    if query_idx is not None:  # pragma: no cover
+        query_idx = int(query_idx)
+        QT_query_idx_allclose = np.allclose(Q, T[query_idx : query_idx + m])
```

Comment:
Now that I think about it further, there's no reason why we can't warn ASAP:

```
    check_window_size(m, max_size=Q.shape[-1])
    if query_idx is not None:  # pragma: no cover
        query_idx = int(query_idx)
        if not np.allclose(Q, T[query_idx : query_idx + m]):
            msg = (
```

and then later:

```
if query_idx is not None: # pragma: no cover
    distance_profile[query_idx] = 0.0
```

I think this is much cleaner

Reply:
@seanlaw 
I think this does not work when there is `np.nan`. In fact, I tried it and we get warning when there is nan. 

For instance, if you try the following  code, you will get two warnings. However, I think users should not se any warning. 

```
T = np.random.rand(10)
m = 3

T[:m] = np.nan
T[-m:] = np.nan
for i in range(len(T) - m + 1):
    Q = T[i:i+m]
    D = core.mass(Q, T, query_idx=i)
```

So, I think your previous proposal, i.e. saving the output of `np.allclose(...)` in variable `QT_query_idx_allclose` and then use it later, is good. I can add a short comment to explain
why we do not add the warning part in the beginning.

Reply:
I think if we want to consider comparing just the finite values of a subsequence, then we should do:

```
Q_idx = np.isfinite(Q) 
T_idx = np.isfinite(T[query_idx : query_idx + m])

if not np.all(Q_idx==T_idx) or not np.allcose(Q[Q_idx], T[T_idx]):
     warning.warn(...)
```

Otherwise, we do not get warning for the following case:

```
Q = [3, np.inf, 1]
T_at_query_idx = [3, 2, np.inf]

```

Reply:
I am cool with that